R/modeldb.R

```diff
@@ -1,17 +1,144 @@
+  message("Done saving the modeldb to ", savefile)
+
+  colDesc <-
+    list(
+      name="Model name that can be used to extract the model from the model library",
+      description="Model description in free from text; in model itself",
+      parameters="A comma separated string listing either the parameter in the model defined by population/individual effects or a population effect parameter",
+      DV="The definition of the dependent variable(s)",
+      filename="Filename of the model.  By default these are installed in the model library and read on demand"
+    )
+  # The names must exactly match
+  stopifnot(all(names(modeldb) %in% names(colDesc)))
+  stopifnot(all(names(colDesc) %in% names(modeldb)))
+  formatText <- sprintf("@format A data frame with %g rows and %g columns", nrow(modeldb), ncol(modeldb))
+  describeText <-
+    sprintf(
+      "\\describe{\n%s}\n",
+      paste(
+        sprintf("  \\item{%s}{%s}\n", names(colDesc), unlist(colDesc)),
+        collapse = ""
+      )
+    )
+  paste(formatText, describeText, sep="\n")
+}
+
+#' Add a directory to the modeldb
+#'
+#' @param dir Directory name containing model files
+#' @param modeldb The starting modeldb data.frame
+#' @return The updated modeldb data.frame
+#' @export
+addDirToModelDb <- function(dir, modeldb=data.frame()) {
+  filesToLoad <-
+    list.files(
+      path = dir,
+      pattern = "\\.R$",
+      ignore.case = TRUE
+    )
+  for (currentFile in filesToLoad) {
+    message("parse currentFile")
+    modeldb <- addFileToModelDb(dir = dir, file = currentFile, modeldb = modeldb)
+  }
+  modeldb
+}
+
+#' @describeIn addDirToModelDb Add a file to the modeldb
+#' @param file The file name (without the directory name)
+#' @export
+addFileToModelDb <- function(dir, file, modeldb) {
+  fileName <- file.path(dir, file)
+
+  # Extract the model from the file
+  parsedFile <- parse(file = fileName)
+  stopifnot(identical(parsedFile[[1]][[1]], as.name("<-")))
+
+  modelName <- as.character(parsedFile[[1]][[2]])
+  packageStartupMessage("Loading ", modelName, " from ", fileName)
+  if (modelName != tools::file_path_sans_ext(file)) {
+    stop("Loading model failed due to filename/modelName mismatch: ", fileName,
+         call.=FALSE) # nocov
+  }
+
+  # Parse the model to get the fixed effects and DV parameters
+  mod <- nlmixr2est::nlmixr(eval(parsedFile))
+
+  description <- mod$meta$description
+  if (is.null(description)) {
+    message("No description for model in ", fileName)
+    description <- NA_character_
+  }
+
+  # Extract the parameter names
+  modParam <- mod$iniDf
+  # Fixed effects
+  modParamFixed <- modParam$name[is.na(modParam$neta1) & is.na(modParam$err)]
+
+  # swap modeled parameter names for the mu-ref parameter names, where
+  # applicable
+  #for (nm in names(mod$getSplitMuModel$pureMuRef)) {
+  #  modParamFixed[modParamFixed %in% nm] <- mod$getSplitMuModel$pureMuRef[[nm]]
+  #}
+
```

Comment:
@mattfidler, The mu-referencing detection seems to be broken.  The error occurs when running `mod$getSplitMuModel$pureMuRef` here.  Do you have any suggestions?